test/events.html

```diff
@@ -1,55 +1,129 @@
 <!doctype html>
 
 <head>
-  <meta charset="UTF-8">
+  <meta charset="utf-8">
   <title>vaadin-date-picker events tests</title>
   <script src="../../web-component-tester/browser.js"></script>
   <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+  <script src="../../iron-test-helpers/mock-interactions.js"></script>
   <link rel="import" href="../../test-fixture/test-fixture.html">
   <link rel="import" href="../vaadin-date-picker.html">
 </head>
 
 <body>
-  <test-fixture id="date-picker">
+  <test-fixture id="datepicker">
     <template>
-      <vaadin-date-picker></vaadin-date-picker>
+      <vaadin-date-picker clear-button-visible></vaadin-date-picker>
     </template>
   </test-fixture>
   <script>
-    describe('events', () => {
-      let datePicker, input;
+  describe('events', () => {
 
-      describe('has-input-value-changed event', () => {
-        beforeEach(async() => {
-          datePicker = fixture('date-picker');
-          input = datePicker._inputElement;
-        });
+    var target;
 
-        it('should fire the event on input value presence change', async() => {
-          const hasInputValueChangedSpy = sinon.spy();
-          datePicker.addEventListener('has-input-value-changed', hasInputValueChangedSpy);
-          input.value = '2000-02-01';
-          input.dispatchEvent(new CustomEvent('input', {bubbles: true, composed: true}));
-          expect(hasInputValueChangedSpy.calledOnce).to.be.true;
+    function inputChar(char) {
+      target.value += char;
+      MockInteractions.keyDownOn(target, char.charCodeAt(0));
+      target.dispatchEvent(new CustomEvent('input', {bubbles: true, composed: true}));
+    }
 
-          hasInputValueChangedSpy.reset();
-          input.value = '';
-          input.dispatchEvent(new CustomEvent('input', {bubbles: true, composed: true}));
-          expect(hasInputValueChangedSpy.calledOnce).to.be.true;
+    function inputText(text) {
+      for (var i = 0; i < text.length; i++) {
+        inputChar(text[i]);
+      }
+    }
+
+    function arrowDown() {
```

Comment:
I would recommend passing the target as an argument. As more tests are created, it can become more difficult to keep track of the target's reference.